scripts/memorystats

```diff
@@ -18,47 +18,126 @@ if [ ! -f "deps/zephyr/scripts/sanitycheck" ]; then
+        echo "Results for ARC" >> "/tmp/memorystats_$OUTNAME.txt"
+        echo "$divider" >> "/tmp/memorystats_$OUTNAME.txt"
+        deps/zephyr/scripts/sanitycheck -z arc/outdir/arduino_101_sss/zephyr.elf >> "/tmp/memorystats_$OUTNAME.txt" 2> /dev/null
+        str=$(tail -2 "/tmp/memorystats_$OUTNAME.txt" | head -1 | sed -e 's/Totals: //g')
+        results="${results}$str\tARC\n"
     else
-        echo "make failed for $f, skipping it"
+        echo "make failed for ARC, skipping it"
     fi
-done
+    # Run test on all the examples
+    for f in $FILES
+    do
+        #filename="${f/samples\//}"
+        filename="${f##*/}"
+        #touch "/tmp/memorystats_$filename"
+        echo "Testing with $filename..."
+        make JS=$f >& /dev/null
+        if [ $? -eq 0 ]; then
+            echo "$divider" >> "/tmp/memorystats_$OUTNAME.txt"
+            echo "Results for $f" >> "/tmp/memorystats_$OUTNAME.txt"
+            echo "$divider" >> "/tmp/memorystats_$OUTNAME.txt"
+            deps/zephyr/scripts/sanitycheck -z outdir/arduino_101/zephyr.elf >> "/tmp/memorystats_$OUTNAME.txt" 2> /dev/null
+            str=$(tail -2 "/tmp/memorystats_$OUTNAME.txt" | head -1 | sed -e 's/Totals: //g')
+            results="${results}$str\t$filename\n"
+        else
+            echo "make failed for $f, skipping it"
+        fi
+    done
 
-# Output the truncated results
-echo ""
-sed -i "1s/^/-= Total values for all demos =-\n$divider\n${results}\n-= Full reports =-\n/" /tmp/memorystats_output.txt
-sed '/Full/Q' /tmp/memorystats_output.txt
+    # Output the truncated results
+    echo ""
+    sed -i "1s/^/-= Total values for all demos =-\n$divider\n${results}\n-= Full reports =-\n/" "/tmp/memorystats_$OUTNAME.txt"
+    #sed '/Full/Q' "/tmp/memorystats_$OUTNAME.txt"
+}
+
+function compare()
+{
+    echo "-= Changes in values since $1 =-"
+    echo "$divider"
+    filenames=$(grep 'js\|ARC' /tmp/memorystats_current.txt | grep bytes | cut -d' ' -f6 | cut -d$'\t' -f2)
+    for file in $filenames
+    do
+        currRom=$(grep $file /tmp/memorystats_current.txt | grep bytes | cut -d' ' -f1)
+        currRam=$(grep $file /tmp/memorystats_current.txt | grep bytes | cut -d' ' -f4)
+        compRom=$(grep $file /tmp/memorystats_$1.txt | grep bytes | cut -d' ' -f1)
+        compRam=$(grep $file /tmp/memorystats_$1.txt | grep bytes | cut -d' ' -f4)
+        currRom="$((currRom - compRom))"
+        currRam="$((currRam - compRam))"
+        echo "ROM use changed $currRom bytes for $file"
+        echo "RAM use changed $currRam bytes for $file"
+        echo "$divider"
+    done
+}
+
+runTests
+
+# If we are not comparing commits, print the current results
+if [[ -z "$numComp" && -z "$nameComp" ]]; then
+    sed '/Full/Q' "/tmp/memorystats_$OUTNAME.txt"
+fi
+
+# If we are comparing results, find the changes and print them
+if [ -n "$numComp" ]; then
+    results=""
+    OUTNAME=$(git log | grep commit | head -$numComp | tail -1 | cut -d' ' -f2)
```

Comment:
A more straightforward syntax than this "-n 10" would be to let users specify git commits like HEAD~10 or HEAD^^^, which all git commits handle and interpret as 10 commits before HEAD or 3 commits before HEAD respectively. You shouldn't have to actually parse such things, just preserve them and pass them on to the git commands.

But as I said in IRC, we don't need to hold up the patch for that, just keep it in mind.